cluster/replication/consumer.go

```diff
@@ -451,26 +451,33 @@ func (c *CopyOpConsumer) processFinalizingOp(ctx context.Context, op ShardReplic
 		return api.ShardReplicationState(""), err
 	}
 
-	do := func() bool {
-		asyncReplicationStatus, err := c.replicaCopier.AsyncReplicationStatus(ctx, op.Op.SourceShard.NodeId, op.Op.TargetShard.NodeId, op.Op.SourceShard.CollectionId, op.Op.SourceShard.ShardId)
-		c.logger.WithFields(logrus.Fields{"err": err, "objects_propagated": asyncReplicationStatus.ObjectsPropagated, "start_diff_time_unix_millis": asyncReplicationStatus.StartDiffTimeUnixMillis, "upper_time_bound_unix_millis": asyncReplicationUpperTimeBoundUnixMillis}).Info("async replication status")
-		return err == nil && asyncReplicationStatus.ObjectsPropagated == 0 && asyncReplicationStatus.StartDiffTimeUnixMillis >= asyncReplicationUpperTimeBoundUnixMillis
-	}
-
 	if ctx.Err() != nil {
 		logger.WithError(ctx.Err()).Debug("error while processing replication operation, shutting down")
 		return api.ShardReplicationState(""), ctx.Err()
 	}
 	// we only check the status of the async replication every 5 seconds to avoid
 	// spamming with too many requests too quickly
+	acceptError := 0
 	err := backoff.Retry(func() error {
-		if ctx.Err() != nil {
-			logger.WithError(ctx.Err()).Debug("error while processing replication operation, shutting down")
-			return backoff.Permanent(ctx.Err())
+		asyncReplicationStatus, err := c.replicaCopier.AsyncReplicationStatus(ctx, op.Op.SourceShard.NodeId, op.Op.TargetShard.NodeId, op.Op.SourceShard.CollectionId, op.Op.SourceShard.ShardId)
+		if err != nil {
+			logger.WithFields(logrus.Fields{
+				"err":                          err,
+				"objects_propagated":           asyncReplicationStatus.ObjectsPropagated,
+				"start_diff_time_unix_millis":  asyncReplicationStatus.StartDiffTimeUnixMillis,
+				"upper_time_bound_unix_millis": asyncReplicationUpperTimeBoundUnixMillis,
+			}).Info("async replication status")
+			acceptError++
+			if acceptError > 3 {
```

Comment:
I'm worried that 3 is too low here (and in dehydrating), my thought is that IIRC for large shard initializing the hashtree on the source and target node may take more than 15 seconds. I don't have data to support this, so could be wrong, but was thinking it might make sense to have this be ~10 or ~20 for now